### PR TITLE
go modules cleanup and remove wrong self import to v1

### PR DESCRIPTION
**Pull request by mcuadros**
This PR's fix some problems with the go modules dependencies:
- Update code to use go-git/go-git.v5 since ` runtime/local/source/git/git.go` was still using the old version.
- Remove a self-dependency of go-micro to v1, I guess by error introducing by the autocomplete tool.

This has removed a lot of dependencies on old packages and for sure reduced the download time. Related to #1641. 

Also I didn't a bit of code improvement on `runtime/local/git/git.go` that is kind of unrelated to the purpose of this PR, let me know if you rather have this PR splitter in two.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -18,24 +23,28 @@ require (
...
 	github.com/google/uuid v1.1.1
 	github.com/gorilla/handlers v1.4.2
+	github.com/gorilla/websocket v1.4.1 // indirect
+	github.com/grpc-ecosystem/go-grpc-middleware v1.1.0 // indirect
+	github.com/grpc-ecosystem/go-grpc-prometheus v1.2.0 // indirect
 	github.com/grpc-ecosystem/grpc-gateway v1.9.5 // indirect
 	github.com/hashicorp/hcl v1.0.0
 	github.com/hpcloud/tail v1.0.0
-	github.com/imdario/mergo v0.3.8
+	github.com/imdario/mergo v0.3.9
```
Comment (domwong) [inline go.mod:43]: This actually causes a break in behaviour causing a failing test. Can you revert this module back to v0.3.8 please ?

**Review by domwong — CHANGES_REQUESTED** [review]
Looks good, just need to revert one of the module updates and retest.

File: go.mod (hunk 1 above)
Comment (vtolstov) [reply go.mod:43]: some deps wants mergo 0.3.9, i'm fixup via replace. but may be we can have better fix

**Comment by vtolstov** [comment]
@mcuadros can you lowe mergo dep in git repo and release next version?

**Comment by mcuadros** [comment]
Maybe if you have a problem with a specific problem you can make a replace

**Comment by vtolstov** [comment]
Do you have specific reason to bump mergo dep? What is the political decision?
Have minimal needed deps or latest released?

**Comment by mcuadros** [comment]
In go-git? Usually, we update to the latest versions. Here ask to `go mod tidy`? :D

**Comment by vtolstov** [comment]
ok, so we pin now prev mergo version, thanks for pr!

**Pull request merged**